nameko_amqp_retry/rpc.py

```diff
@@ -0,0 +1,71 @@
+import sys
+
+from nameko.messaging import HeaderEncoder, QueueConsumer
+from nameko.rpc import Rpc as NamekoRpc
+from nameko.rpc import RpcConsumer as NamekoRpcConsumer
+
+from nameko_amqp_retry import Backoff, BackoffPublisher
+from nameko_amqp_retry.constants import (
+    CALL_ID_STACK_HEADER_KEY, RPC_METHOD_ID_HEADER_KEY)
+
+
+class RpcConsumer(NamekoRpcConsumer):
+
+    def handle_message(self, body, message):
+
+        # use the rpc_method_id if set, otherwise fall back to the routing key
+        method_id = message.headers.get(RPC_METHOD_ID_HEADER_KEY)
```

Comment:
RPC can be routed via a header, not just the routing key :) ok, should be noted in "hacker's guide to Nameko" :)

is this because the retry needs to be redelivered with dead-lettering via a default exchange and then end up in the targeted queue again?